### PR TITLE
CSC Unpacker fix for handling of rare CSC data corruption

### PR DESCRIPTION
**Pull request by barvic**
CSC Unpacker fix for handling of rare data corruption, related to invalid CSC DMB header/trailer data

#### PR description:

- related to HLT crash in run-367770 from CSCDCCUnpacker issue #41747 in CMSSW_13_0_6
- observed single HLT crash case so far
- was reproduced with supplied by HLT script and data file
- added safeguard against invalid nullptr data, caused by rare case of CSC hardware readout corruption

#### PR validation:

Ran PR validation runTheMatrix basic battery of tests.
Notes:
Supplied by the HLT test script for 13_0_6 doesn't work with 13_2_X and 13_1_X (doesn't look like CSC related issue), 
but the script passed without crashes with this code modification for 13_0_X.
For 13_2_X and 13_1_X tests was using custom minimal CSC test script to run only affected muonCSCDigis CSC unpacking sequence with corrupted data file.
Tests passed without crashes with proposed modification.

This PR would need backport for 13_1_X and 13_0_X.

### CODE REVIEW

**Comment by fwyzard** [comment]
Is it understood what causes `csc_itr->second` to be `nullptr` in the first place ?

**Comment by barvic** [comment]
@fwyzard 
Current understanding is that CSC readout electronics board (single DMB in this particular case) was in failed state and sent corrupted data chunk (basically garbage).
That data chunk was containing corrupted data words combinations, which by coincidence were similar to signature data words (in this case DMB trailer words), which are used to mark and locate data chunks of particular readout boards in the event raw data buffer.
So this chunk was still considered data format-wise as a chamber data candidate for unpacking recovery attempt and error reporting, 
and then was passed for further unpacking, but with NULL pointer to that invalid chamber data chunk position in the FED event data buffer. 
That chunk is unrecoverable and should be rejected from the unpacking. 
Added nullptr  check should let skip unpacking of such corrupted data.

General comment:
Current CSC unpacking policy is to try to extract as much as possible data for multiple chambers from single FED event data, especially in cases of partial readout data corruption, caused by individual chambers/boards. It lets us to unpack CSC data more efficiently, but rarely we could still observe cases of new previously unknown data corruptions, which could potentially cause unpacking issues, like this one and which are handled on case by case basis.
(As far as I recall previous case of unknown severe data corruption, which was causing  CSC unpacking issues was back in 2015.)

**Comment by fwyzard** [comment]
@barvic thank you for the explanation!

File: EventFilter/CSCRawToDigi/src/CSCDDUEventData.cc [hunk 1]
```diff
@@ -193,6 +193,14 @@ void CSCDDUEventData::unpack_data(const uint16_t* buf, CSCDCCExaminer* examiner)
         if (cscid != -1) {
           const uint16_t* pos = (const uint16_t*)csc_itr->second;
 
+          if (pos == nullptr) {
+            if (debug)
+              LogTrace("CSCDDUEventData|CSCRawToDigi")
+                  << "skip unpacking of CSC " << cscid << " due format errors (NULL pointer to chamber data)"
+                  << std::dec;
```
Comment (missirol) [inline EventFilter/CSCRawToDigi/src/CSCDDUEventData.cc:200]:
```suggestion
                  << "skip unpacking of CSC " << cscid << " due to format errors (NULL pointer to chamber data)"
                  << std::dec;
```
Just a small typo. One question: is the idea to print `cscid` as "hex", and then switch to `std::dec` ?

Comment (barvic) [reply EventFilter/CSCRawToDigi/src/CSCDDUEventData.cc]: @missirol thanks! (cut-n-paste legacy typos from other log message few lines below). Fixed both.

**Comment by missirol** [comment]
please test

**Comment by missirol** [comment]
type bugfix

@cms-sw/reconstruction-l2 , could you please review this simple fix ? It addresses the HLT issue in #41747.

@barvic, could you please backport this PR to `CMSSW_13_1_X` and `CMSSW_13_0_X` ? (if it helps, I can also take care of the backports, just let me know)

**Comment by mandrenguyen** [comment]
+reconstruction

**Pull request merged**